qiskit_dynamics/dispatch/wrap.py

```diff
@@ -18,86 +18,30 @@
 from .array import Array
 
 
-def wrap(func: Callable, wrap_return: bool = True, decorator: bool = False) -> Callable:
+def wrap(
+    func: Callable, wrap_return: bool = True, wrap_args: bool = True, decorator: bool = False
+) -> Callable:
     """Wrap an array backend function to work with Arrays.
```

Comment:
Can you expand on this to more explicitly say how it sets up the function to "work with arrays"? I.e. add a sentence or two explaining that it unwraps `Array`-type arguments (or anything with `__qiskit_array__`) and wraps the results.